src/charm.py

```diff
@@ -26,12 +26,60 @@ def __init__(self, *args):
         # event observations
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.stop, self._on_stop)
+        self.framework.observe(
+            self.on['elasticsearch'].relation_changed,
+            self._on_elasticsearch_relation_changed
+        )
+        self.framework.observe(
+            self.on['elasticsearch'].relation_broken,
+            self._on_elasticsearch_relation_broken
+        )
+
+        # initialized stored variables
+        # TODO: test whether we can just pass the {ingress-address}:{port} string
+        #       from Elasticsearch to Graylog or if we need to send multiple hosts
+        #       Hypothesis: just the ingress is fine
+        self._stored.set_default(elasticsearch_uri=str)  # connection str for elasticsearch
+
+    @property
+    def has_elasticsearch(self):
+        if self._stored.elasticsearch_uri:
+            return True
+        else:
+            return False
 
     def _on_config_changed(self, _):
         self._configure_pod()
 
-    def _on_stop(self, event):
-        self.unit.status = MaintenanceStatus('Pod is terminating')
+    def _on_stop(self, _):
+        self.unit.status = MaintenanceStatus('Pod is terminating.')
+
+    def _on_elasticsearch_relation_changed(self, event):
+        """Get the relation data from the relation and save to stored variable."""
+        # skip if unit is not leader
+        if not self.unit.is_leader():
+            return
+
+        data = event.relation.data[event.unit]
+        ingress_address = data.get('ingress-address')
+        port = data.get('port')
+        if ingress_address is None or port is None:
+            logger.warning('No port or ingress-address in Elasticsearch relation data')
+            return
+
+        # if we have the data we need, get the information
+        self._stored.elasticsearch_uri = 'http://{}:{}'.format(
```

Comment:
Are we going to have issues if `elasticsearch_uri` is over https?

Reply:
This is definitely something we need to keep in mind in the future, but the Elasticsearch charm currently does not have TLS enabled and I know it isn't a top priority as of now.